src/App.tsx

```diff
@@ -1,26 +1,154 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { useEffect } from 'react';
+import { Route, Routes, useNavigate } from 'react-router-dom';
+import { StoreState } from './redux-toolkit/store';
+import { useDispatch, useSelector } from 'react-redux';
+import { setAccessToken, setExpirationTime, setId, setIsLoggedIn, setRole } from './redux-toolkit/features/user/user-slice';
+import { AdminRoutes, HrRoutes, StudentRoutes } from './utils/protected-routes';
+import axios from 'axios';
+import jwtDecode from 'jwt-decode';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+import { LoginView } from './pages/LoginView/LoginView';
+import { RegisterView } from './pages/RegisterView/RegisterView';
+import { ForgotPasswordView } from './pages/ForgotPasswordView/ForgotPasswordView';
+import { ChangePasswordView } from './pages/ChangePasswordView/ChangePasswordView';
+import { HrHomeView } from './pages/HrHomeView/HrHomeView';
+import { HrSelectedStudentsView } from './pages/HrSelectedStudentsView/HrSelectedStudentsView';
+import { HrStudentProfileView } from './pages/HrStudentProfileView/HrStudentProfileView';
+import { StudentView } from './pages/StudentView/StudentView';
+import { HrProfileView } from './pages/HrProfileView/HrProfileView';
+import { HomeAdmin } from './components/HomeAdmin/HomeAdmin';
+import { apiUrl } from './config/api';
+
+export const App = () => {
+  const { expirationTime, role } = useSelector((store: StoreState) => store.user);
+  const dispatch = useDispatch();
+  let navigate = useNavigate();
+  const axiosJWT = axios.create();
+
+  const refreshToken = async () => {
+    try {
+      const res = await fetch(`${apiUrl}/refresh-token`, {
```

Comment:
Myślę, że dosyć często się to będzie powtarzać. Można by to przenieść do funkcji, służącej do komunikacji z API.